src/Bicep.Core.Samples/Files/baselines/InvalidResourceDerivedTypes_LF/main.bicep

```diff
@@ -0,0 +1,29 @@
+type invalid1 = resource
+
+type invalid2 = resource<>
+
+type invalid3 = resource<'abc', 'def'>
+type invalid4 = resource<hello>
+type invalid5 = resource<'Microsoft.Storage/storageAccounts'>
+type invalid6 = resource<'Microsoft.Storage/storageAccounts@'>
+type invalid7 = resource<'Microsoft.Storage/storageAccounts@hello'>
+type invalid8 = resource<'notARealNamespace:Microsoft.Storage/storageAccounts@2022-09-01'>
+type invalid9 = resource<':Microsoft.Storage/storageAccounts@2022-09-01'>
+type invalid10 = resource<'abc' 'def'>
+type invalid11 = resource<123>
+type invalid12 = resource<resourceGroup()>
+
+type thisIsWeird = resource</*
+*/'Astronomer.Astro/organizations@2023-08-01-preview' 
+///  >
+>
+
+type shouldWeBlockThis = resource<'Microsoft.${'Storage'}/storageAccounts@2022-09-01'>
```

Comment:
This was unexpected - I'm wondering if we should support the same behavior as we have for `resource` declarations

Reply:
I was unsure about this... ideally, I think we should support using simple expressions like:

```bicep
var resourceType = 'Microsoft.Storage/storageAccounts@2022-09-01'
type saType = resource<resourceType>
```

Probably safer to block it for now. Also, the completions in this PR assume resource type names have to be string literals

Reply:
> I was unsure about this... ideally, I think we should support using simple expressions like:
> 
> ```bicep
> var resourceType = 'Microsoft.Storage/storageAccounts@2022-09-01'
> type saType = resource<resourceType>
> ```
> 
> Probably safer to block it for now. Also, the completions in this PR assume resource type names have to be string literals

Sounds good. Agree that it would be useful, but I'd want to introduce this capability holistically (e.g. with #444)